SixTrack/sixtrack.s

```diff
@@ -3883,50 +3883,66 @@
+     &sin((((sigmv(j)/((clight*(e0f/e0))))*                             &!hr03
+     &crabfreq)*2d0)*pi + crabph(ix)))*oidpsv(j)                         !hr03
+      ejv(j)=ejv(j) -                                                   &!hr03
+     &((((((crabamp*crabfreq)*2d0)*pi)/clight)                          &!hr03
+     &*xv(xory,j))*                                                     &!hr03
+     &cos((((sigmv(j)/(clight)*crabfreq)*2d0)*pi                        &!hr03
+     & + crabph(ix)))                                                    !hr03
 +ei
 +ei
 +if tilt
 +if crlibm
-        yv(xory,j)=yv(xory,j) - crabamp*                                &!hr03
-     &sin_rn((((sigmv(j)/clight)*crabfreq)*2d0)*pi + crabph(ix))         !hr03
-      dpsv(j)=dpsv(j) -                                                 &!hr03
-     &((((((crabamp*crabfreq)*2d0)*pi)/clight)*xv(xory,j))*             &!hr03
-     &cos_rn((((sigmv(j)/clight)*crabfreq)*2d0)*pi + crabph(ix)))*c1m3   !hr03
+       yv(xory,j)=yv(xory,j) - crabamp*c1e3/e0f*                        &!hr03
+     &sin_rn((((sigmv(j)/((clight*(e0f/e0))))*                          &!hr03
+     &crabfreq)*2d0)*pi + crabph(ix))*oidpsv(j)                          !hr03
+      ejv(j)=ejv(j) -                                                   &!hr03
+     &((((((crabamp*crabfreq)*2d0)*pi)/clight)                          &!hr03
+     &*xv(xory,j))*                                                     &!hr03
+     &cos_rn((((sigmv(j)/(clight*(e0f/e0)))*crabfreq)*2d0)*pi           &!hr03
+     & + crabph(ix)))                                                    !hr03
 +ei
 +if .not.crlibm
-        yv(xory,j)=yv(xory,j) - crabamp*                                &!hr03
-     &sin((((sigmv(j)/clight)*crabfreq)*2d0)*pi + crabph(ix))            !hr03
-      dpsv(j)=dpsv(j) -                                                 &!hr03
-     &((((((crabamp*crabfreq)*2d0)*pi)/clight)*xv(xory,j))*             &!hr03
-     &cos((((sigmv(j)/clight)*crabfreq)*2d0)*pi + crabph(ix)))*c1m3      !hr03
+       yv(xory,j)=yv(xory,j) - crabamp*c1e3/e0f*                        &!hr03
+     &sin((((sigmv(j)/((clight*(e0f/e0))))*                             &!hr03
+     &crabfreq)*2d0)*pi + crabph(ix))*oidpsv(j)                          !hr03                                    
+      ejv(j)=ejv(j) -                                                   &!hr03
+     &((((((crabamp*crabfreq)*2d0)*pi)/clight)                          &!hr03
+     &*xv(xory,j))*                                                     &!hr03
+     &cos((((sigmv(j)/(clight*(e0f/e0)))*crabfreq)*2d0)*pi              &!hr03
+     & + crabph(ix)))                                                    !hr03
 +ei
 +ei
       ejf0v(j)=ejfv(j)
-      ejfv(j)=dpsv(j)*e0f+e0f
-      ejv(j)=sqrt(ejfv(j)**2+pma**2)                                     !hr03
+      ejfv(j)=sqrt(ejv(j)**2-pma**2)                                     !hr03
```

Comment:
Probably you have already done, these steps should be a carbon copy of the code after the normal cavity

Reply:
Was this checked?

Also, I would think that x**2-y**2 is not as accurate as (x-y)(x+y), since you are then subtracting or adding slightly less similar-but-large-numbers